### PR TITLE
Forward Qt logging to Rust log!() facade

### PR DESCRIPTION
**Pull request by ratijas**
This is an implementation of proposal #85.

### CODE REVIEW

**Comment by ratijas** [comment]
I'd like to implement coercion of Qt/Rust logging levels using From/Into traits before merge :thinking:

File: qmetaobject/Cargo.toml [hunk 1]
```diff
@@ -16,6 +16,7 @@ chrono_qdatetime = ["chrono"]
 [dependencies]
 qmetaobject_impl = { path = "../qmetaobject_impl", version = "=0.1.4"}
 lazy_static = "1.0"
+log = "0.4"
```
Comment (ogoffart) [inline qmetaobject/Cargo.toml:19]: can it be made an optional dependency?

File: qmetaobject/src/log.rs [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+/// Mapping from Qt logging levels to Rust logging facade's levels.
+///
+/// Due to the limited range of levels from both sides,
+/// [`QtCriticalMsg`][`QtMsgType`] and [`QtFatalMsg`][`QtMsgType`]
+/// both map to [`log::Level::Error`][Level],
+/// while [`log::Level::Trace`][Level] is never returned.
+///
+/// [Level]: https://docs.rs/log/0.4.10/log/enum.Level.html
+/// [`QtMsgType`]: https://doc.qt.io/qt-5/qtglobal.html#QtMsgType-enum
+pub fn map_level(lvl: QtMsgType) -> Level {
```
Comment (ogoffart) [inline qmetaobject/src/log.rs:85]: I wouldn't put these two function `pub`
The From  impls should be enough.

File: qmetaobject/Cargo.toml (hunk 1 above)
Comment (ogoffart) [reply qmetaobject/Cargo.toml:19]: Then it should be documented in the crate's top level documentation in the list of features

Comment (ratijas) [reply qmetaobject/Cargo.toml:19]: See https://github.com/woboq/qmetaobject-rs/issues/85#issuecomment-622176232 for my rationale.

But then again, yes, of course I could make it optional.

File: qmetaobject/src/log.rs (hunk 2 above)
Comment (ratijas) [reply qmetaobject/src/log.rs]: OK, reverted back to `From`.

File: qmetaobject/Cargo.toml [hunk 3]
```diff
@@ -11,13 +11,15 @@ keywords = ["Qt", "QML", "QMetaObject",]
 repository = "https://github.com/woboq/qmetaobject-rs"
 
 [features]
+default = ["log"]
```
Comment (ogoffart) [inline qmetaobject/Cargo.toml:14]: Thanks.  I'm just concerned about having too many dependencies.  That's also why i wouldn't put it as the default either.

File: README.md [hunk 4]
```diff
@@ -58,9 +59,33 @@ Window {
 
 Requires Qt >= 5.8
 
+## Cargo features
+
+Cargo provides a way to enable (or disable default) optional [features](https://doc.rust-lang.org/cargo/reference/features.html).
+
+### `log`
+
+By default, Qt's logging system is not initialized, and messages from e.g. QML's `console.log` don't go anywhere.
```
Comment (ogoffart) [inline README.md:68]: Don't they go in the stderr by default?

I mean, they follow the default Qt handler which is also quite configurable with env variable: https://doc.qt.io/qt-5/debug.html#warning-and-debugging-messages  /  https://doc.qt.io/qt-5/qloggingcategory.html#configuring-categories

Comment (ratijas) [reply README.md:68]: Idk, by default they just not showing up at all on my system.

File: qmetaobject/Cargo.toml (hunk 3 above)
Comment (ratijas) [reply qmetaobject/Cargo.toml:14]: I think debugging is essential, especially on such multi-lingua projects, and should be on by default. By having it on by default AND providing instructions for activation on the front-page example we are lowering the barrier for newcomers.

**Comment by ratijas** [comment]
Breaking change incoming:  3da8fe7

**Pull request merged**